src/components/product/Product.jsx

```diff
@@ -0,0 +1,61 @@
+import { useCart } from '../../context/cart-context';
+import './Product.css';
+
+export default function Product(props) {
+    const { _id, url, title, author, price, discount, categoryName } = props.product;
+
+    const { store, dispatch } = useCart();
+
+    const getDiscountedPrice = (price, discount) => {
+        return Number(price - ((price * discount) / 100));
+    }
+
+    const isItemInCart = (id) => {
+        const { itemsInCart } = store;
+        return itemsInCart.includes(id);
+    }
+
+    const isItemInWishlist = (id) => {
```

Comment:
I liked how you have created this utility function. Try to move it in diff folder the code will be more clean